File: CPP/PROGRAM/Keyboard.cpp
#pragma once
#include <iostream>
#include <string>
#include "Peripheral.cpp"

using namespace std;

class Keyboard : public Peripheral{
    //atribut
    private:
        string jenisSwitch;
        int RGB;

    public:
        //konstruktor
        Keyboard(){
            this->jenisSwitch = "";
            this->RGB = 0;
        }

        Keyboard(string jenisSwitch, int RGB, string tipePeripheral, string interface, string merk, string nama) : Peripheral(tipePeripheral, interface, merk, nama){
            this->jenisSwitch = jenisSwitch;
            this->RGB = RGB;
        }

        //getter dan setter
        void setJenisSwitch(string jenisSwitch){
            this->jenisSwitch = jenisSwitch;
        }

        void setRGB(int RGB){
            this->RGB = RGB;
        }

        string getJenisSwitch(){
            return this->jenisSwitch;
        }

        int getRGB(){
            return this->RGB;
        }

        ~Keyboard(){

        }
};

File: PYTHON/PROGRAM/Keyboard.py
from Peripheral import Peripheral

class Keyboard(Peripheral):
    #konstruktor
    def __init__(self, tipePeripheral = "", interface = "", merk = "", nama = "", jenisSwitch = "", RGB = 0):
        super().__init__(merk, nama, tipePeripheral, interface)
        self.__jenisSwitch = jenisSwitch
        self.__RGB = RGB

    #getter dan setter
    def getJenisSwitch(self):
        return self.__jenisSwitch
    
    def setJenisSwitch(self, jenisSwitch):
        self.__jenisSwitch = jenisSwitch

    def getRGB(self):
        return self.__RGB
    
    def setRGB(self, RGB):
        self.__RGB = RGB